src/composables/useEmailSubscription.ts

```diff
@@ -1,92 +1,248 @@
-import sign, { SubscribeType } from '@/helpers/sign';
-import { getInstance } from '@snapshot-labs/lock/plugins/vue3';
+import sign, { DataType } from '@/helpers/sign';
+import { JSONRPCSuccess, JSONRPCError } from '@/helpers/interfaces';
 
-export function useEmailSubscription() {
-  enum Status {
-    waiting,
-    success,
-    error
+let isInitialized = false;
+let isSubscriptionsLoading = false;
+
+const SubscribeSchema: DataType = {
+  Subscribe: [
+    { name: 'address', type: 'address' },
+    { name: 'email', type: 'string' }
+  ]
+};
+
+const UpdateSubscriptionsSchema: DataType = {
+  Subscriptions: [
+    { name: 'address', type: 'address' },
+    { name: 'email', type: 'string' },
+    { name: 'subscriptions', type: 'string[]' }
+  ]
+};
+
+enum Status {
+  waiting,
+  success,
+  error
+}
+
+enum Level {
+  info = 'info',
+  warning = 'warning',
+  'warning-red' = 'warning-red',
+  success = 'success'
+}
+
+const subscriptionTypes = ['summary', 'newProposal', 'closedProposal'] as const;
+type SubscriptionType = (typeof subscriptionTypes)[number];
+
+const status: Ref<Status> = ref(Status.waiting);
+const postSubscribeLevel: Ref<Level> = ref(Level.info);
+const postSubscribeMessage = ref('');
+const loading = ref(false);
+const isSubscribed = ref(false);
+const shouldRemoveEmail = ref(true);
+
+const { t } = useI18n();
+const { web3Account } = useWeb3();
+const { setAlias, aliasWallet, isValidAlias, checkAlias } = useAliasAction();
+const apiSubscriptions: Ref<SubscriptionType[]> = ref([]);
+const clientSubscriptions = computed({
+  get() {
+    return subscriptionTypes.reduce((acc, type) => {
+      acc[type] = apiSubscriptions.value.includes(type);
+      return acc;
+    }, {} as Record<SubscriptionType, boolean>);
+  },
+  set(value) {
+    apiSubscriptions.value = Object.entries(value)
+      .map(([key, value]) => (value ? key : undefined))
+      .filter(Boolean)
+      .map(key => key as SubscriptionType);
   }
+});
 
-  enum Level {
-    info = 'info',
-    warning = 'warning',
-    'warning-red' = 'warning-red',
-    success = 'success'
+async function signWithAlias(message, typesSchema) {
```

Comment:
You should be using `actionWithAlias` method in `useAliasAction` instead of repeating the logic here again